api/envoy/config/core/v3/health_check.proto

```diff
@@ -69,15 +69,24 @@ message HealthCheck {
     option (udpa.annotations.versioning).previous_message_type =
         "envoy.api.v2.core.HealthCheck.Payload";
 
+    enum DecoderType {
+      HEX = 0;
+      UTF8 = 1;
+    }
+
     oneof payload {
```

Comment:
@htuch IMHO the use of Payload as text comparator is not so straight-forward. The `Payload` type is highly targeted towards binary comparisons. If one looks at the current state it seems that there are 2 modes: text and binary, but actually there should be 3 modes: text-plain, text-hex, binary.
Would it be better if we explicitly supply 3 states: text, binary, text-plain?

Reply:
There could also be Base64 encode text maybe in future, and so on. I am fine with ether the current approach or use multiple oneof text fields.

Reply:
I'm confused. Why do we need oneoff between bytes and string? bytes should be able to cover all use case where string may be used.

Reply:
Thanks Yan, talked offline. It makes sense that UTF8 string can be actually carried by the bytes binary.  We can get rid of DecoderType. We will just keep the text HEX usage as TCP did before.  HTTP can use HEX text and bytes binary.

Reply:
Can you check the most recent changes? Thank you all!